core/src/main/java/com/netease/arctic/utils/ConvertStructUtil.java

```diff
@@ -70,14 +73,76 @@ public static com.netease.arctic.ams.api.DataFile convertToAmsDatafile(
       });
       amsDataFile.setUpperBounds(amsUpperBounds);
     }
-    DefaultKeyedFile.FileMeta fileMeta = DefaultKeyedFile.parseMetaFromFileName(dataFile.path().toString());
-    amsDataFile.setFileType(fileMeta.type().name());
-    amsDataFile.setIndex(fileMeta.node().index());
-    amsDataFile.setMask(fileMeta.node().mask());
-    amsDataFile.setFileType(fileMeta.type().name());
+
+
+    /*
+    Iceberg file has 3 types(FileContent) : DATA, POSITION_DELETES, EQUALITY_DELETES
+    Arctic file has 4 types(DataFileType): BASE_FILE, INSERT_FILE, EQ_DELETE_FILE, POS_DELETE_FILE 
+    i.  for iceberg DATA file, arctic keyed table has 3 file type: BASE_FILE, INSERT_FILE, EQ_DELETE_FILE;
+        and arctic unkeyed table has 1 file type: BASE_FILE
+    ii. for iceberg POSITION_DELETES file, arctic file type is POS_DELETE_FILE
+    iii.for iceberg EQUALITY_DELETES file, arctic is unsupported now
+     */
+    if (table.isKeyedTable()) {
+      FileContent content = dataFile.content();
+      if (content == FileContent.DATA) {
+        DefaultKeyedFile.FileMeta fileMeta = FileUtil.parseKeyedFileMetaFromFileName(dataFile.path().toString());
+        validateArcticFileType(content, dataFile.path().toString(), fileMeta.type());
+        amsDataFile.setFileType(fileMeta.type().name());
+        amsDataFile.setIndex(fileMeta.node().index());
+        amsDataFile.setMask(fileMeta.node().mask());
+      } else if (content == FileContent.POSITION_DELETES) {
+        DefaultKeyedFile.FileMeta fileMeta = FileUtil.parseKeyedFileMetaFromFileName(dataFile.path().toString());
+        amsDataFile.setFileType(DataFileType.POS_DELETE_FILE.name());
+        if (fileMeta.type() == DataFileType.POS_DELETE_FILE) {
+          amsDataFile.setIndex(fileMeta.node().index());
+          amsDataFile.setMask(fileMeta.node().mask());
+        } else if (fileMeta.type() == DataFileType.BASE_FILE) {
```

Comment:
It seems `FileUtil.parseKeyedFileMetaFromFileName` use node(0, 0) as its default value, so maybe you can use this default value here instead of setting specific node(0, 0)

Reply:
Done. Now both POS_DELETE_FILE and BASE_FILE use the default (0,0) from fileMeta.